pkg/server/adminHandler.go

```diff
@@ -314,7 +334,14 @@ func (h *adminHandler) deviceConfigSet(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 	var deviceConfig config.EdgeDevConfig
-	err = json.Unmarshal(body, &deviceConfig)
+	switch r.Header.Get(contentType) {
+	case mimeProto:
+		err = proto.Unmarshal(body, &deviceConfig)
+	case mimeJSON:
+		fallthrough
+	default:
+		err = json.Unmarshal(body, &deviceConfig)
```

Comment:
I think someday it would be good if we could make the `content-type` header required, so the default is a 400, but this is fine for now.

Can we make it explicit, rather than `fallthrough`? Those always lead to problems when someone doesn't realize the order and changes it.